core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -928,4 +929,23 @@ mod tests {
         assert_eq!(scheduling_summary.num_unschedulable, 0);
         assert_eq!(collect_work(&work_receivers[0]).1, vec![vec![2], vec![0]]);
     }
+
+    #[test]
+    fn test_schedulable_threads() {
+        let (mut scheduler, _work_receivers, _finished_work_sender) = create_test_frame(2);
+        scheduler.config.max_scheduled_cus = 2; // only allow 1 tx per thread
```

Comment:
this option doesn't exist in 2.1/2.0, so not easy to backport the test, since the limit is just based on a constant.